nuxeo-platform-spreadsheet-web-ui/src/main/resources/web/nuxeo.war/ui/nuxeo-spreadsheet/nuxeo-spreadsheet-button.html

```diff
@@ -0,0 +1,139 @@
+<!--
+(C) Copyright 2016 Nuxeo SA (http://nuxeo.com/) and others.
+
+Licensed under the Apache License, Version 2.0 (the License);
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an AS IS BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Contributors:
+   Adilio Araujo <aaraujo@nuxeo.com>
+-->
+
+<!--
+`nuxeo-spreadsheet-button`
+@group Nuxeo UI
+@element nuxeo-spreadsheet-button
+-->
+<dom-module id="nuxeo-spreadsheet-button">
+
+  <template>
+    <style>
+      :host {
+        display: block;
+      }
+
+      paper-icon-button {
+        background: white;
+        color: var(--nuxeo-action-color);
+        border: 1px solid var(--divider-color);
+        border-radius: 4px;
+      }
+
+      .dialog {
+        width: 100%;
+        height: 100%;
+        padding: 0;
+      }
+
+      #iframe {
+        width: 100%;
+        height: 100%;
+        margin: 0;
+        padding: 0;
+      }
+    </style>
+
+    <template is="dom-if" if="[[_isAvailable(nxProvider, mode)]]">
+      <paper-icon-button id="button" icon="editor:mode-edit" on-tap="_show"></paper-icon-button>
+      <paper-tooltip for="button">[[i18n('spreadsheetButton.tooltip')]]</paper-tooltip>
+    </template>
+
+    <paper-dialog id="dialog" class="dialog" with-backdrop>
+      <iframe id="iframe" frameborder="0" scrolling="auto" on-load="_onLoad"></iframe>
+    </paper-dialog>
+
+  </template>
+
+  <script>
+    Polymer({
+      is: 'nuxeo-spreadsheet-button',
+      behaviors: [Nuxeo.RoutingBehavior, Nuxeo.I18nBehavior, Nuxeo.FiltersBehavior],
+      properties: {
+        baseUrl: {
+          type: String,
+          value: '/nuxeo/'
+        },
+        mode: {
+          type: String,
+          value: ''
+        },
+        nxProvider: {
+          type: HTMLElement,
+          value: null
+        }
+      },
+
+      _isAvailable: function() {
+        return (this.nxProvider !== null) && this.mode === 'table';
+      },
+
+      _show: function() {
+        // convert from nxProvider.sort to sortInfos
+        var sortInfos = [];
+        Object.keys(this.nxProvider.sort).forEach(function(key) {
+          sortInfos.push({'sortColumn': key, 'sortAscending': this.nxProvider.sort[key] === 'asc'});
+        }.bind(this));
+
+        // convert nxProvider.params and nxProvider.aggregations to properties
+        var properties = {};
+        Object.keys(this.nxProvider.params).forEach(function(key) {
+          properties[key] = this.nxProvider.params[key];
+        }.bind(this));
```

Comment:
if you do a `var provider = this.nxProvider;` then you don't have to keep binding all the functions to `this` just to access `this.nxProvider`